### PR TITLE
Replace SDB wtih HT in librz/core/agraph.c

### PR DESCRIPTION
**Pull request by p371jah**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Review by XVilka — APPROVED** [review]
Hah, SDB was used as hashtable before. Good catch! :+1:

File: librz/core/agraph.c [hunk 1]
```diff
@@ -920,23 +916,30 @@ static int is_valid_pos(const RzAGraph *g, int l, int pos) {
...
+		for (int j = 0; j < g->layers[i].n_nodes; j++) {
 			RzGraphNode *gn = g->layers[i].nodes[j];
-			const RzList *Ln = hash_get_rlist(res, gn);
+			const RzList *Ln = ht_pp_find(res, gn, NULL);
 			const RzANode *an = get_anode(gn);
 
 			if (!Ln) {
 				RzList *vert = rz_list_new();
-				hash_set(res, gn, vert);
+				ht_pp_update(res, gn, vert);
```
Comment (ret2libc) [inline librz/core/agraph.c]: This is just a `ht_pp_insert`, isn't it?

File: librz/core/agraph.c [hunk 2]
```diff
@@ -1146,36 +1156,43 @@ static void place_nodes(const RzAGraph *g, const RzGraphNode *gn, int is_left, S
 	}
 
 	graph_foreach_anode (lv, itk, gk, ak) {
-		hash_set(res, gk, p);
-		hash_set(placed, gk, true);
+		ht_pu_update(res, gk, (ut64)(size_t)p);
```
Comment (ret2libc) [inline librz/core/agraph.c:1161]: is also this just an insert? I think we should use `_insert` when we want to do just an insert.

File: librz/core/agraph.c [hunk 3]
```diff
@@ -1146,36 +1156,43 @@ static void place_nodes(const RzAGraph *g, const RzGraphNode *gn, int is_left, S
...
 	if (!classes) {
 		return NULL;
 	}
 
-	Sdb *res = sdb_new0();
-	Sdb *placed = sdb_new0();
+	HtPUOptions pu_opt = { 0 };
+	HtPPOptions pp_opt = { 0 };
+	HtPU *res = ht_pu_new_opt(&pu_opt);
+	SetP *placed = (SetP *)ht_pp_new_opt(&pp_opt);
```
Comment (ret2libc) [inline librz/core/agraph.c:1178]: this is risky.. But I see that setp was designed only for strings, which is IMHO wrong. A Set based on strings should be called like `SetS` or `SetString` or something like that. But I guess that's ok for now :(

**Comment by ret2libc** [comment]
With this PR I get these warnings that are not present in dev:
```
WARNING: hash_get_int: assertion 'found' failed (line 1035)
WARNING: hash_get_int: assertion 'found' failed (line 1035)
WARNING: hash_get_int: assertion 'found' failed (line 1035)
```
This happens while working on ./test/bins/elf/ls, with the `main` function.

File: librz/core/agraph.c (hunk 2 above)
Comment (p371jah) [reply librz/core/agraph.c:1161]: No, overrides are possible here.

**Comment by p371jah** [comment]
@ret2libc You can achieve same result on dev by replacing `hash_get_int()` macro with with the below snippet. Anyway I have removed this assertion.
```c
static int hash_get_int(Sdb *sdb, const void *k) {
	ut32 cas;
	int val = sdb_num_get(sdb, sdb_fmt("%" PFMT64u, (ut64)(size_t)(k)), &cas);
	rz_return_val_if_fail(cas != 0, 0);
	return val;
}
```

**Review by ret2libc — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Pull request merged**